### PR TITLE
Da/django doctor

### PR DESCRIPTION
**Pull request by davidpofo**
Some changes made that a tool called django doctor mentioned. Issues include:

- Hard-coded URL in template
- hard-coded static asset URL in template
- nullable string field
- foreignkey missing related_name
- model method order

### CODE REVIEW

File: siteapp/urls.py [hunk 1]
```diff
@@ -11,19 +11,15 @@
 from .settings import *
 
 urlpatterns = [
-    url(r"^$", views.homepage, name="homepage"),
-    url(r"^login$", views.homepage, name="homepage"),
-    url(r"^(privacy|terms-of-service|love-assessments)$", views.shared_static_pages),
+    url(r"^(?![\s\S]|login)$", views.homepage, name="homepage"),
+    url(r"^(privacy|terms-of-service|love-assessments)$", views.shared_static_pages, name="privacy_terms_love"),
```
Comment (omarabed15) [inline siteapp/urls.py:15]: I have no clue how these seemingly unrelated items all use the same shared view and pages, but hey... that's out of scope of this PR.

File: templates/systems/controls_selected.html [hunk 2]
```diff
@@ -58,23 +58,18 @@ <h2 class="control-heading">
...
-            <li><a href="/systems/{{ system.id }}/controls/baseline/NIST_SP-800-53_rev5/moderate/_assign">Assign 800-53 rev5 Moderate Impact Baseline</a></li>
-            <li><a href="/systems/{{ system.id }}/controls/baseline/NIST_SP-800-53_rev5/high/_assign">Assign 800-53 rev5 High Impact Baseline</a></li>
-            -->
-            <li><a href="/systems/{{ system.id }}/controls/baseline/NIST_SP-800-171_rev1/cui/_assign">Assign 800-171 rev1 Baseline</a></li>
+            <li><a href={% url 'assign_baseline' system_id=system.id catalog_key='NIST_SP-800-53_rev4' baseline_name='low' %}>Assign 800-53 rev4 Low Impact Baseline</a></li>
+            <li><a href={% url 'assign_baseline' system_id=system.id catalog_key='NIST_SP-800-53_rev4' baseline_name='moderate' %}>Assign 800-53 rev4 Moderate Impact Baseline</a></li>
+            <li><a href={% url 'assign_baseline' system_id=system.id catalog_key='NIST_SP-800-53_rev4' baseline_name='high' %}>Assign 800-53 rev4 High Impact Baseline</a></li>
+            <li><a href={% url 'assign_baseline' system_id=system.id catalog_key='NIST_SP-800-171_rev1' baseline_name='cui' %}>Assign 800-171 rev1 Baseline</a></li>
```
Comment (omarabed15) [inline templates/systems/controls_selected.html:64]: So much cleaner and clearer. Great job!

**Review by omarabed15 — APPROVED** [review]
Looks great!

File: siteapp/urls.py (hunk 1 above)
Comment (davidpofo) [reply siteapp/urls.py:15]: good question so the view shared_static_pages uses the page name as an arg so the url tag in the template is looking for a page arg and the url pattern is only supplying one either privacy or terms-of-service or love-assessments)

**Pull request merged**